authority-portal-frontend/src/app/pages/catalog-page/catalog-page/catalog-page.component.ts

```diff
@@ -156,9 +166,39 @@ export class CatalogPageComponent implements OnInit, OnDestroy {
   }
 
   onDataOfferClick(dataOffer: CatalogDataOffer) {
+    this.openDataOfferDialog(dataOffer.assetId, dataOffer.connectorId);
+  }
+
+  private openDataOfferDialog(assetId: string, connectorId: string) {
     this.assetDetailDialogService
-      .open(dataOffer.assetId, dataOffer.connectorId, this.ngOnDestroy$)
+      .open(assetId, connectorId, this.ngOnDestroy$)
+      .pipe(
+        finalize(() => {
+          this.changeUrlToCatalogRoot();
+        }),
+      )
       .subscribe();
+
+    this.router.navigate([connectorId, assetId], {
+      relativeTo: this.route,
+      queryParams: {
+        environmentId: this.route.snapshot.queryParams.environmentId,
+      },
+    });
+    // BreadcrumbService builds the name from the URL which is nonsensical in casse of asset IDs
+    document.title = 'MDS Catalog - Data Offer';
+  }
+
+  private changeUrlToCatalogRoot() {
+    const currentRoute = this.route.snapshot;
+    this.router.navigate(
+      [currentRoute.url.map((segment) => segment.path).join('/')],
+      {
+        queryParams: {
+          environmentId: currentRoute.queryParams.environmentId,
+        },
```

Comment:
isnt there a simpler way to do this? like just hardcoding the catalog path?

Reply:
Unfortunately, the component is also used in the route `my-organization/data-offers` and needs to have the same functionality. This is why I need to build the url this way